### PR TITLE
ci(py/dotpromptz): add smoke test to verify dotpromptz usability from pypi

### PR DESCRIPTION
**Pull request by MengqinShen**
CHANGELOG:
1. After publishing dotpromptz to pypi, a smoke test added to verify its usability. 
2. Add supported Python version test in smoke test( Single run of build job can create a universal wheel (py3-none-any.whl) and 3.10, 3.11, 3.12, 3.13 (Full Matrix) were tested)

### CODE REVIEW

File: .github/workflows/publish_python_dotpromptz_package.yml [hunk 1]
```diff
@@ -112,7 +114,7 @@ jobs:
       - name: Upload build packages
         uses: actions/upload-artifact@v4
         with:
-          name: dist-dotpromptz
+          name: dist-dotpromptz-${{ matrix.python-version }}-${{ github.run_attempt }}
```
[the remark this answers is not in the document]
Comment (MengqinShen) [reply .github/workflows/publish_python_dotpromptz_package.yml]: @yesudeep Now I just figure it out.  In check_and_build (Build Job), we only use to build the pure Python dotpromptz package with Python 3.12  with a single run, but it creates a universal wheel (py3-none-any.whl). This single artifact works for all Python 3 versions. Later in smoke test, we  run across python version matrix, to ensure that the published package is usable by every supported Python version.
